crates/compiler/can/src/env.rs

```diff
@@ -161,35 +178,46 @@ impl<'a> Env<'a> {
                             .collect();
                         Err(RuntimeError::ValueNotExposed {
                             module_name: self
-                                .module_ids
+                                .qualified_module_ids
                                 .get_name(module_id)
                                 .expect("Module ID known, but not in the module IDs somehow")
+                                .as_inner()
                                 .clone(),
                             ident: Ident::from(ident),
                             region,
                             exposed_values,
                         })
                     }
                 },
-                None => Err(RuntimeError::ModuleNotImported {
-                    module_name: self
-                        .module_ids
-                        .get_name(module_id)
-                        .expect("Module ID known, but not in the module IDs somehow")
-                        .clone(),
-                    imported_modules: self
-                        .dep_idents
-                        .keys()
-                        .filter_map(|module_id| self.module_ids.get_name(*module_id))
-                        .map(|module_name| module_name.as_ref().into())
-                        .collect(),
-                    region,
-                    module_exists: true,
-                }),
+                _ => Err(self.module_exists_but_not_imported(scope, module_id, region)),
             }
         }
     }
 
+    #[inline]
```

Comment:
I'd suggest leaving the `inline` annotation off and letting the compiler decide - it's usually pretty smart about it. We've run into some issues before where we inlined too aggressively and caused debug builds to blow up because the stack frames grow quite large.

https://roc.zulipchat.com/#narrow/stream/231635-compiler-development-.28private.29/topic/casual.20conversation/near/305880702